File: Core/Shared/JavaLib/src/net/lukx/jchatter/lib/models/Message.java
package net.lukx.jchatter.lib.models;

import net.lukx.jchatter.lib.PublicApi;

import java.util.Date;

/***
 * Class used for deserialization of server data
 */
@SuppressWarnings("unused")
@PublicApi
public class Message {
    /**
     * ID
     */
    public int id;
    /***
     * ID of user who sent this message.
     * FK to User's ID
     */
    public int idsender;
    /***
     * Id of room that this was sent to.
     * FK to Room's ID
     */
    public int idroomReceiver;
    /***
     * Content
     */
    public String content;
    /***
     * If message has been received by anyone
     */
    public boolean received;
    /***
     * If message has been seen by anyone
     */
    public boolean seen;
    /***
     * If message ever was edited
     */
    public boolean edited;
    /***
     * If the user is currently writing this message
     */
    public boolean writing;
    /***
     * When this message was sent
     */
    public Date dateSent;
    /***
     * When this message was received
     */
    public Date dateReceived;
    /***
     * When this message was first read
     */
    public Date dateRead;
    /***
     * When this message was last edited
     */
    public Date dateEdited;

    public Message(){

    }

}

File: Core/Wpf/ChatterWpf/ChatterWpf/Models/Message.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace ChatterWpf.Models
{

    public class Message
    {

        public int id{get;set;}

        //FK to User's ID, ID of user who sent this message.

        public int idsender{get;set;}

        public int idroomReceiver{get;set;}

        public string content{get;set;}

        public bool received{get;set;}

        public bool seen{get;set;}

        public bool edited{get;set;}

        public bool writing{get;set;}

        public DateTime? dateSent{get;set;}

        public DateTime? dateReceived{get;set;}

        public DateTime? dateRead{get;set;}

        public DateTime? dateEdited{get;set;}

        public Message()
        {

        }
    }

}